extensions/KHR/SPV_KHR_16bit_storage.asciidoc

```diff
@@ -0,0 +1,261 @@
+from a 32-bit type.
+ - A structure containing a 16-bit member can be an operand to <<OpArrayLength, *OpArrayLength*>>.
+ - Any other instructions not explicitly listed by the capabilities or allowed by the validations rules 
+   cannot operate on variables with 16-bit scalar, 16-bit vector, or 16-bit composite types. 
+
+A *FPRoundingMode* decoration can be applied only to:
+
+ - a width-only conversion instruction that is used as the object argument of an
+ <<OpStore, *OpStore*>> storing through a pointer to a 16-bit floating-point
+ object in *Uniform*, or *PushConstant*, or *Input*, or *Output*
+ <<Storage_Class,Storage Classes>>.
+
+Interactions with SPV_KHR_storage_buffer_storage_class
+------------------------------------------------------
+
+If https://gitlab.khronos.org/spirv/spirv-extensions/blob/master/SPV_KHR_uniform_buffer_storage_class.asciidoc[SPV_KHR_uniform_buffer_storage_class] is supported, ::
+
+modify the description of the *StorageUniformBufferBlock16* capability, adding the following sentence to the first paragraph of the description: :::
+
+The object can also be in the *StorageBuffer* <<Storage_Class,Storage Class>> and have any decorations supported for this <<Storage_Class,Storage Class>>.
+
+modify the description of the *StorageUniform16* capability, adding the following sentence to the first paragraph of the description: :::
+
+The object can also be in the *StorageBuffer* <<Storage_Class,Storage Class>> and have any decorations supported for this <<Storage_Class,Storage Class>>.
+
+Modify Section 3.31, Capability, adding the following rows to the Capability table: ::
+
+[options="header"]
+|====
+2+^| Capability ^| Depends On
+| 4433 | *StorageBuffer16BitAccess* +
+Same as *StorageUniformBufferBlock16* |
+| 4434 | *UniformAndStorageBuffer16BitAccess* +
+Same as *StorageUniform16*| *StorageBuffer16BitAccess*
+|====
+
+Issues
+------
+
+Revision History
+----------------
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1|2016-11-22|Alexander Galazin|*Initial revision*
+|2|2016-11-28|Alexander Galazin|*Address first round of feedback*
+|3|2016-12-01|Alexander Galazin|*Removed combined Load/Store and Convert instructions.
+Renamed capabilities and described them in terms of storage classes.*
+|4|2016-12-08|David Neto|*Assigned token numbers*
+|5|2016-12-14|Alexander Galazin| *Renamed the extension. Removed changes to the default rounding modes. Made StorageUniform16 dependent on StorageUniformBufferBlock16*
+|6|2017-02-22|JohnK| *Clarified that conversions for changing width can only change the width, not the fundamental type domain.*
+|7|2017-03-15|Alexander Galazin| *Clarified that FP Rounding mode can be used only if the capabilities from this extension are enabled*
+|8|2017-03-23|Alexander Galazin| Added interactions with *SPV_KHR_uniform_buffer_storage_class*
+|9|2018-06-11|Alexander Galazin| Added clarifications for SPIR-V issue 319
```

Comment:
@johnkslang did you also check that the HTML files are up to date? https://www.khronos.org/registry/spir-v/extensions/KHR/SPV_KHR_16bit_storage.html is out of date (is missing revision 9), and I don't see it included in this change.

Reply:
I did, and in fact 3 were out of date, which is what the 3rd commit fixed.

But, this branch and master both show that revision 9 is already in place.  I wonder about your link.  If I navigate to the normal "blob" type view from the top, for either master or tidy-up branches, I see the revision 9 change in the extension.

So, I'm thinking there is some cache problem with that non-blob link.

Reply:
OK, I see that the github version is up to date. It is just the khronos.org mirror that is stale. Unfortunately, the khronos.org mirror is the first hit on Google, at least for extensions old enough to be present there.

The links at https://www.khronos.org/registry/spir-v/ all point to github now. Maybe the khronos.org mirror should just be deleted, and then google would stop linking to them? @outofcontrol do you control the mirror on khronos.org or know the history here?